### PR TITLE
type: add miss generic of Upload prop

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

https://github.com/ant-design/ant-design/blob/decfac3540791abc9a9b166bd411ce5cfd6b65f6/components/upload/interface.ts#L56-L58

这里应该和上面是保持一致的：

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | type: add miss generic of Upload prop |
| 🇨🇳 Chinese | type: 补充 Upload 组件缺失的泛型 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
来个 type test

**Comment by li-jia-nan** [comment]
> 来个 type test

done~

File: components/upload/__tests__/type.test.tsx [hunk 1]
```diff
@@ -210,4 +211,14 @@ describe('Upload.typescript', () => {
...
     expect(upload3).toBeTruthy();
   });
+
+  it('UploadListProps type', () => {
+    const props: UploadListProps<number | string> = {
+      locale: {},
+      removeIcon: (file) => <span>{file.response}</span>,
+      downloadIcon: (file) => <span>{file.response}</span>,
+      previewIcon: (file) => <span>{file.response}</span>,
+    };
```
Comment (afc163) [inline components/upload/__tests__/type.test.tsx:221]: 给 https://github.com/ant-design/ant-design/pull/48612 也加一下？

Comment (li-jia-nan) [reply components/upload/__tests__/type.test.tsx:231]: done~

**Review by afc163 — APPROVED** [review]

**Pull request merged**